### ISSUE TITLE
Custom Chart: Simplified gradient visualization

### ISSUE THREAD

**Issue by miraodasilva** [wandb/client#1537]
Hi everyone,

Once again, I'd like to say I'm a big fan of this tool, and the more I use it the more I like it. Regarding weight, bias and gradient visualization, I checked some guides and saw that using log="all" in wand.init allows you to visualize these values over time. However, in my case (and my network is not even that large), I get 224 weights and gradients to look through. This is ok, but not really viable for me since going through all 224 plots is simply not time efficient. Is there a way to visualize these values in a more wholistic way, like in a single/couple plot(s)? Please let me know if I am missing something.

Thanks.

**Comment by ariG23498** [comment]
Hey @miraodasilva 
Are you talking about **224** individual plots for weights, bias and gradient visualization? This means that you are actually working with `224 * 3 = 672` plots all together.
With a wholistic approach of visualization could you elaborate on what you mean here?

I am not certain about your way of approaching the problem. If there are 224 individual models, then your present state would make sense to me. But if you are using a sweep configuration or something else where you are comparing the grads, bias and weights together, grouping the jobs and visualizing the plots by overlaying them would make sense here. 
Reference for Grouping: https://docs.wandb.com/library/grouping
Reference for Sweeps: https://docs.wandb.com/app/features/sweeps

**Comment by miraodasilva** [comment]
Hi, really sorry for the late response.

I have 232 parameter plots (weights and biases) and corresponding 232 gradient plots (weights and biases). This corresponds to every single weight and bias in my (moderately sized, nothing colossal) model. My point was that if I want to check if gradients are flowing properly it is effectively impossible to do so at a glance. Going through 232 plots is feasible as a last resort but in my opinion it is not really that useful for everyday scenarios. I would prefer to have, say, a plot where the value of the gradient is the y-axis and the weight+biases are enumerated in the x-axis. This is implemented in a somewhat crude form in the following  [discussion](https://discuss.pytorch.org/t/check-gradient-flow-in-network/15063/7) . I have used this in the past: 232 ticks in the x-axis is not the most readable, but it kind of works decently at a glance as a heartrate monitor for gradient flow in a network.

Perhaps I am asking too much, and this is not within the scope of w&b. Or perhaps I am missing something and there _is_ a way to group all 232 graphs into a readable plot on w&b. Please let me know if you have any suggestions. Thanks a lot.

**Comment by cvphelps** [comment]
Hi Rodrigo, I see your suggestion on how to visualize these more effectively. Would you be interested in trying our Custom Charts feature to explore your logged gradients?

https://docs.wandb.com/app/features/custom-charts

Something like this, from your linked [discussion](https://discuss.pytorch.org/t/check-gradient-flow-in-network/15063/7) might be possible.

![image](https://user-images.githubusercontent.com/6355078/100694372-ca03bc80-3343-11eb-984e-374a11d1d768.png)

**Comment by miraodasilva** [comment]
@cvphelps Sure, I will try to implement this ASAP and let you know. Thanks.

**Comment by ariG23498** [comment]
Hey @miraodasilva 
If `custom charts` solved your problem would you let us know so that we can close this ticket?
:smile:

**Comment by miraodasilva** [comment]
No worries, I will close it. Currently I am bit busy but once I implement I will post it here, might be useful for someone else/as a potential feature. Thanks a lot.

**Comment by ierezell** [comment]
Hello, 

Sorry to bump this old issue but it's a feature that I would also like to have. 

Is there any example of a custom chart to plot the "whole layers gradient average" as a sanity check for the model? 
As usually when a model is stale it's the whole X hundred plots that have a small gradient so the average would be quite informative of the general health of the training. 

Or is there any plan to have it automatically (like current gradients)? 

Thanks in advance
Have a great day.

**Comment by amorehead** [comment]
Hi, all. I second @ierezell's comment. I think a per-layer gradient plot could be extremely useful for folks, including myself.
